web/yo/app/scripts/filters/Filter.js

```diff
@@ -1,127 +1,135 @@
+                return item !== '$$hashKey';
+            }).sort(function(a, b) {
+                if (delayAttrs.indexOf(a) !== -1) {
+                    return 1;
+                } else if (delayAttrs.indexOf(b) !== -1) {
+                    return -1;
+                } else if (prioAttrs.indexOf(a) !== -1) {
+                    return -1;
+                } else if (prioAttrs.indexOf(b) !== -1) {
+                    return 1;
+                } else if (a < b) {
+                    return -1;
+                }
+                return 1;
+            });
+            return keys.map(function(key) {
+                if (typeof obj[key] !== 'object') {
+                    var _obj = {};
+                    Object.defineProperty(_obj, '$value', {
+                        enumerable: false,
+                        value: obj[key]
+                    });
+                    obj[key] = _obj;
+                }
+                return Object.defineProperty(obj[key], '$key', {
+                    enumerable: false,
+                    value: key
+                });
+            });
+        };
+    })
+    .filter('sortPubs', function() {
+        return function(array) {
+            if (angular.isArray(array)) {
+                array.sort(function(a, b) {
+                    var strA = '';
+                    var strB = '';
+                    var numA = -1;
+                    var numB = -1;
 
-        strA = a.type.trim();
-        strB = b.type.trim();
-        numA = a.id.trim();
-        numB = b.id.trim();
+                    strA = a.type.trim();
+                    strB = b.type.trim();
+                    numA = a.id.trim();
+                    numB = b.id.trim();
 
-        if(strA !== strB) {
-          return strA < strB;
-        }else{
-          return numA < numB;
-        }
-      });
-    }
-    return array;
-  };
-})
-.filter('typeaheadFilter', function (Levenshtein) {
-    return function (input, query) {
-        var result = [];
+                    if (strA !== strB) {
+                        return strA < strB;
+                    }
+                    return numA < numB;
+                });
+            }
+            return array;
+        };
+    })
+    .filter('typeaheadFilter', function(Levenshtein) {
+        return function(input, query) {
+            var result = [];
 
-        query = query.toString().toLowerCase();
-        angular.forEach(input, function (object) {
-            if (object.toLowerCase().indexOf(query.toLowerCase()) !== -1) {
-                result.push(object);
+            query = query.toString().toLowerCase();
+            angular.forEach(input, function(object) {
+                if (object.toLowerCase().indexOf(query.toLowerCase()) !== -1) {
+                    result.push(object);
+                }
+            });
+            result.sort(function(a, b) {
+                a = a.toString().toLowerCase();
+                b = b.toString().toLowerCase();
```

Comment:
Not sure about what data type the result item is. If it is object, should we use JSON.stringify(a)? since a.toString() will return "[object Object]"